default.nix

```diff
@@ -1,157 +1,92 @@
-    inherit packageFun cargo rustc;
-    packageResolver = { source, name, version, sha256, ... }:
       {
-        src = resolver { inherit source name version; };
+        rustcflags = {
+          "registry+https://github.com/rust-lang/crates.io-index"."*" = [ "--cap-lints" "warn" ];
+        };
+        nativeBuildInputs = {
+          "registry+https://github.com/rust-lang/crates.io-index".curl-sys."*" = darwinFrameworks;
+          "registry+https://github.com/rust-lang/crates.io-index".libgit2-sys."*" = darwinFrameworks;
+        };
+        buildInputs = {
+          "registry+https://github.com/rust-lang/crates.io-index".libgit2-sys."*" = [ pkgs.libiconv ];
+          "registry+https://github.com/rust-lang/crates.io-index".cargo."*" = [ pkgs.libiconv ] ++ darwinFrameworks;
+          unknown.cargo2nix2."*" = [ pkgs.libiconv ] ++ darwinFrameworks;
+        };
+        environment = {
+          "registry+https://github.com/rust-lang/crates.io-index".openssl-sys."*" =
+            let
+              envize = s: builtins.replaceStrings ["-"] ["_"] (lib.toUpper s);
+              envBuildPlatform = envize pkgs.buildPlatform.config;
+              envHostPlatform = envize pkgs.hostPlatform.config;
+              staticOpenssl = pkgs: (pkgs.openssl.override {
+                # `perl` is only used at build time, but the derivation incorrectly uses host `perl` as an input.
+                perl = pkgs.buildPackages.buildPackages.perl;
+                static = true;
+              }).overrideAttrs (drv: {
+                installTargets = "install_sw";
+                outputs = [ "dev" "out" "bin" ];
+                postInstall = builtins.replaceStrings ["rm -r " "rmdir "] ["rm -rf " "rm -rf "] drv.postInstall;
+              });
+
+              joinOpenssl = openssl: buildPackages.symlinkJoin {
+                name = "openssl"; paths = with openssl; [ out dev ];
+              };
+            in
+              # We don't use key literals here, as they might collide if `hostPlatform == buildPlatform`.
+              builtins.listToAttrs [
+                { name = "${envBuildPlatform}_OPENSSL_DIR"; value = joinOpenssl (staticOpenssl buildPackages); }
+                { name = "${envHostPlatform}_OPENSSL_DIR"; value = joinOpenssl (staticOpenssl pkgs); }
+                { name = "OPENSSL_STATIC"; value = "1"; }
+              ];
+        };
       };
-    excludeCrates.unknown = "*";
-    environment.OPENSSL_DIR = openssl pkgs;
-    nativeBuildInputs = [
-      pkgs.buildPackages.buildPackages.jq
-      pkgs.cacert # This is added as a workaround for https://github.com/target/lorri/issues/98
-    ];
 
-    inherit (rustPackages.config) features;
+  rustPkgs = pkgs.rustBuilder.makePackageSet {
+    inherit cargo rustc packageFun rustPackageConfig;
+    buildRustPackages = buildPackages.rustBuilder.makePackageSet {
+      inherit cargo rustc packageFun rustPackageConfig;
+    };
   };
-}
+in
+  rustPkgs."cargo2nix2 0.1.0 unknown" { }
```

Comment:
It seems that this pull request completely removes the `shell` feature, the automatic vendoring into the Nix store with `.cargo/config` source replacement, and also airplane mode. Do you intend to put this functionality back in?

Reply:
Yeah, as we're not actively using this functionality, I plan to bring it back later.

Reply:
That's fair! But since this is a large breaking change, we should update the README to reflect this and open an issue to reintroduce the functionality later, if we are planning to merge this pull request directly into `master` without adding it back.